### ISSUE TITLE
[bug] When switching, it will ask you to switch networks twice

### ISSUE THREAD

**Issue by paulflex87** [rainbow-me/rainbowkit#1134]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### RainbowKit Version

0.12.5

### wagmi Version

0.12.8

### Current Behavior

When I change networks, for example from mainnet to polygon, after confirming the change to polygon, metamask activates again and asks to change the network to goerli

### Expected Behavior

I hope you don't get a request to change to goerli network, because I didn't ask for it

### Steps To Reproduce

Click connect button
```
import { ConnectButton } from "@rainbow-me/rainbowkit";

function ProfileComponent() {
    return <ConnectButton/>
}
```

rainbowkit configuration:
```
import "@rainbow-me/rainbowkit/styles.css";

import { getDefaultWallets, RainbowKitProvider } from "@rainbow-me/rainbowkit";
import { configureChains, createClient, WagmiConfig } from "wagmi";
import { mainnet, polygon, optimism, arbitrum, sepolia,goerli } from "wagmi/chains";
import { alchemyProvider } from "wagmi/providers/alchemy";
import { publicProvider } from "wagmi/providers/public";

const { chains, provider } = configureChains(
  [mainnet,polygon,optimism, sepolia],
  [
    publicProvider(),
  ]
);

const { connectors } = getDefaultWallets({
  appName: "My app",
  chains,
});

const wagmiClient = createClient({
  autoConnect: true,
  connectors,
  provider,
});

export { WagmiConfig, RainbowKitProvider, wagmiClient, chains };

```

last part file:
```
ReactDOM.createRoot(document.getElementById("root") as HTMLElement).render(
  <React.StrictMode>
      <WagmiConfig client={wagmiClient}>
        <RainbowKitProvider chains={chains}>
          <App />
        </RainbowKitProvider>
      </WagmiConfig>
  </React.StrictMode>
);
```

**Comment by DanielSinclair** [comment]
This appears to be an issue in the `10.28.1` release of MetaMask. I was able to reproduce on Uniswap, and am seeing these console errors. Uniswap uses `web3-react` so this issue is likely unrelated to RainbowKit/wagmi.
```
MetaMask - RPC Error: networkConfigurationId [object Object] does not match a configured networkConfiguration 
{code: -32603, message: 'networkConfigurationId [object Object] does not match a configured networkConfiguration', data: {
    "code": -32603,
    "message": "networkConfigurationId [object Object] does not match a configured networkConfiguration",
    "data": {
        "originalError": {}
    }
}
```

**Comment by DanielSinclair** [comment]
@paulflex87 It looks like these issues were fixed with [v10.28.2](https://github.com/MetaMask/metamask-extension/releases/tag/v10.28.2) and [v10.28.3](https://github.com/MetaMask/metamask-extension/releases/tag/v10.28.3). Let me know if you're still bumping into them!
